apps/site/lib/site_web/plugs/route.ex

```diff
@@ -20,4 +20,9 @@ defmodule SiteWeb.Plugs.Route do
         assign(conn, :route, route)
     end
   end
+
+  def call(conn, []) do
+    # Do nothing, there is no route to add
+    conn
```

Comment:
This was added because the endpoint `/stops/:stop_id/schedules` doesn't have a `route_id` and calls to it were failing at this plug.  The controller this endpoint calls uses the plug (`ScheduleController.schedules_for_stop/2`)

I put the `schedules_for_stop/2` function in the `schedules_controller.ex` because I am trying to consolidate code instead of making a new controller for every end point.  To me it makes sense to organize based off data model types (`schedules_for_stop/2` returns a list of schedules which is why it's in the schedule controller).  It could be added to its own controller but I feel like that adds to the clutter of controllers we already have.

Adding this case for the plug allowed the endpoint to work, and the 404 error is still rendered if a bad `route_id` is passed to any function that calls this plug.

Reply:
Thanks for explaining! My concern is if this plug should only be used where a `route` parameter is required. It seems that there might be other use cases where raising an exception when the `route` parameter is not set would be the desired behavior. Perhaps instead within `schedules_controller` you could fine-tune which functions call this plug. 
For example from the [plug docs](https://hexdocs.pm/phoenix/plug.html#controller-plugs), something like
plug `SiteWeb.Plugs.Route` when action not in [:schedules_for_stop]`